cmake does not check whether uuid-dev or libssl-dev installed

```
root@jdsf-test:~/jdcloud-sdk-cpp-signer# sudo dpkg -P libssl-dev
(Reading database ... 149624 files and directories currently installed.)
Removing libssl-dev:amd64 (1.0.2g-1ubuntu4.15) ...
root@jdsf-test:~/jdcloud-sdk-cpp-signer# sudo dpkg -P uuid-dev
(Reading database ... 149538 files and directories currently installed.)
Removing uuid-dev:amd64 (2.27.1-6ubuntu3.6) ...
root@jdsf-test:~/jdcloud-sdk-cpp-signer# cmake .
-- Configuring done
-- Generating done
-- Build files have been written to: /root/jdcloud-sdk-cpp-signer
root@jdsf-test:~/jdcloud-sdk-cpp-signer# make
Scanning dependencies of target jdcloudsigner
[  4%] Building CXX object src/CMakeFiles/jdcloudsigner.dir/JdcloudSigner.cpp.o
[  8%] Building CXX object src/CMakeFiles/jdcloudsigner.dir/JdcloudSignerImpl.cpp.o
/root/jdcloud-sdk-cpp-signer/src/JdcloudSignerImpl.cpp:22:23: fatal error: uuid/uuid.h: No such file or directory
compilation terminated.
src/CMakeFiles/jdcloudsigner.dir/build.make:86: recipe for target 'src/CMakeFiles/jdcloudsigner.dir/JdcloudSignerImpl.cpp.o' failed
make[2]: *** [src/CMakeFiles/jdcloudsigner.dir/JdcloudSignerImpl.cpp.o] Error 1
CMakeFiles/Makefile2:146: recipe for target 'src/CMakeFiles/jdcloudsigner.dir/all' failed
make[1]: *** [src/CMakeFiles/jdcloudsigner.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/README.md
+++ b/README.md
@@ -45,7 +45,7 @@
 ### Linux（Ubuntu）
 1. 安装开发依赖库
 ```
-sudo apt-get install g++ cmake libssl-dev uuid-dev
+sudo apt-get install g++ cmake pkg-config libssl-dev uuid-dev
 ```
 2. 从GitHub下载Demo例子，地址为：https://github.com/jdcloud-api/jdcloud-sdk-cpp-signer
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -20,6 +20,12 @@ set (Demo_VERSION_DEBUG 0)
 set(VERSION ${Demo_VERSION_MAJOR}.${Demo_VERSION_MINOR}.${Demo_VERSION_DEBUG})
 set(SOVERSION ${VERSION})
 
+if(APPLE)
+    set(PKGCONFIG_REQUIRES_PRIVATE "libcrypto")
+else(APPLE)
+    set(PKGCONFIG_REQUIRES_PRIVATE "libcrypto uuid")
+endif(APPLE)
+
 CONFIGURE_FILE(
     ${CMAKE_CURRENT_SOURCE_DIR}/libjdcloud_signer.pc.in
     ${CMAKE_CURRENT_BINARY_DIR}/libjdcloud_signer.pc
@@ -41,7 +47,7 @@ elseif(APPLE)
     link_libraries(OpenSSL::Crypto)
 elseif(UNIX)
     find_package(PkgConfig REQUIRED)
-    pkg_check_modules(depends REQUIRED libcrypto uuid)
+    pkg_check_modules(depends REQUIRED ${PKGCONFIG_REQUIRES_PRIVATE})
     link_libraries(${depends_LIBRARIES})
     include_directories(${depends_INCLUDE_DIRS})
 endif()

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,4 +1,4 @@
-# Copyright 2018 JDCLOUD.COM
+# Copyright 2018,2019 JDCLOUD.COM
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -20,12 +20,6 @@ set (Demo_VERSION_DEBUG 0)
 set(VERSION ${Demo_VERSION_MAJOR}.${Demo_VERSION_MINOR}.${Demo_VERSION_DEBUG})
 set(SOVERSION ${VERSION})
 
-if(APPLE)
-    set(PKGCONFIG_REQUIRES_PRIVATE "libcrypto")
-else(APPLE)
-    set(PKGCONFIG_REQUIRES_PRIVATE "libcrypto uuid")
-endif(APPLE)
-
 CONFIGURE_FILE(
     ${CMAKE_CURRENT_SOURCE_DIR}/libjdcloud_signer.pc.in
     ${CMAKE_CURRENT_BINARY_DIR}/libjdcloud_signer.pc
@@ -46,9 +40,10 @@ elseif(APPLE)
     FIND_PACKAGE(OpenSSL)
     link_libraries(OpenSSL::Crypto)
 elseif(UNIX)
-    include_directories(/usr/local/ssl/include)
-    link_libraries(crypto)
-    link_libraries(uuid)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(depends REQUIRED libcrypto uuid)
+    link_libraries(${depends_LIBRARIES})
+    include_directories(${depends_INCLUDE_DIRS})
 endif()
 
 aux_source_directory(. DIR_LIB_SRCS)

